From e46036958f1632f1cbf0e7fc79b45e8bb20fa901 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Hugo=20Beauz=C3=A9e-Luyssen?= <hugo.beauzee@datadoghq.com>
Date: Mon, 17 Feb 2025 08:00:40 +0100
Subject: [PATCH] remove inclusions prior to Python.h

stdbool.h will cause <features.h> to be included before Python.h can
define some macros to enable some additional features, causing multiple
types not to be defined down the line
---
 Objects/codeobject.c    | 2 --
 Python/assemble.c       | 3 +--
 Python/compile.c        | 4 ++--
 Python/legacy_tracing.c | 1 -
 Python/pythonrun.c      | 4 ++--
 5 files changed, 5 insertions(+), 9 deletions(-)

diff --git a/Objects/codeobject.c b/Objects/codeobject.c
index 1be66215638..4b5f33983ab 100644
--- a/Objects/codeobject.c
+++ b/Objects/codeobject.c
@@ -1,5 +1,3 @@
-#include <stdbool.h>
-
 #include "Python.h"
 #include "opcode.h"
 #include "structmember.h"         // PyMemberDef
diff --git a/Python/assemble.c b/Python/assemble.c
index 8789d8ef978..4aa922848f2 100644
--- a/Python/assemble.c
+++ b/Python/assemble.c
@@ -1,11 +1,10 @@
-#include <stdbool.h>
-
 #include "Python.h"
 #include "pycore_code.h"          // write_location_entry_start()
 #include "pycore_compile.h"
 #include "pycore_opcode.h"        // _PyOpcode_Caches[] and opcode category macros
 #include "pycore_pymem.h"         // _PyMem_IsPtrFreed()
 
+#include <stdbool.h>
 
 #define DEFAULT_CODE_SIZE 128
 #define DEFAULT_LNOTAB_SIZE 16
diff --git a/Python/compile.c b/Python/compile.c
index 7255f5d1475..699878269bc 100644
--- a/Python/compile.c
+++ b/Python/compile.c
@@ -21,8 +21,6 @@
  * objects.
  */
 
-#include <stdbool.h>
-
 #include "Python.h"
 #include "pycore_ast.h"           // _PyAST_GetDocString()
 #define NEED_OPCODE_TABLES
@@ -56,6 +54,8 @@
  */
 #define STACK_USE_GUIDELINE 30
 
+#include <stdbool.h>
+
 #undef SUCCESS
 #undef ERROR
 #define SUCCESS 0
diff --git a/Python/legacy_tracing.c b/Python/legacy_tracing.c
index 4a6565bebc1..570f140a9c7 100644
--- a/Python/legacy_tracing.c
+++ b/Python/legacy_tracing.c
@@ -2,7 +2,6 @@
  * Provides callables to forward PEP 669 events to legacy events.
  */
 
-#include <stddef.h>
 #include "Python.h"
 #include "opcode.h"
 #include "pycore_ceval.h"
diff --git a/Python/pythonrun.c b/Python/pythonrun.c
index 5f3d249df45..44b76f545ea 100644
--- a/Python/pythonrun.c
+++ b/Python/pythonrun.c
@@ -8,8 +8,6 @@
 
 /* TODO: Cull includes following phase split */
 
-#include <stdbool.h>
-
 #include "Python.h"
 
 #include "pycore_ast.h"           // PyAST_mod2obj
@@ -27,6 +25,8 @@
 #include "errcode.h"              // E_EOF
 #include "marshal.h"              // PyMarshal_ReadLongFromFile()
 
+#include <stdbool.h>
+
 #ifdef MS_WINDOWS
 #  include "malloc.h"             // alloca()
 #endif
-- 
2.34.1

